os/kernel/binary_manager/binary_manager_bootparam.c

```diff
@@ -294,42 +301,55 @@ void binary_manager_update_bootparam(int requester_pid, uint8_t type)
 		for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
 			/* Scan binary files */
 			ret = binary_manager_check_user_update(bin_idx, &check_update);
-			if (ret == OK && check_update) {
+			if (ret == BINMGR_OK) {
 				/* Update index for inactive partition */
 				update_bp_data.app_data[BIN_BPIDX(bin_idx)].useidx ^= 1;
 				need_update = true;
+			} else if (ret == BINMGR_ALREADY_UPDATED || ret == BINMGR_NOT_FOUND) {
+				bmdbg("No binary to update: bin_idx %d, ret %d\n", bin_idx, ret);
+			} else {
+				bmdbg("Failed to check user update: bin_idx %d, ret %d\n", bin_idx, ret);
+				goto send_response;
 			}
 		}
 		if (!need_update) {
-			bmdbg("No binary to update\n");
-			goto send_response;
+			bmdbg("No App binaries to update\n");
+			is_all_updatable = false;
+			response_msg.data.result[BINARY_USERAPP] = BINMGR_ALREADY_UPDATED;
 		}
```

Comment:
Why do you need this section separately? For kernel and common, you added them in `else  if`.

Reply:
That's because there may be multiple user binaries.